src/bentoml/_internal/utils/__init__.py

```diff
@@ -303,6 +304,24 @@ def resolve_user_filepath(filepath: str, ctx: t.Optional[str]) -> str:
     raise FileNotFoundError(f"file {filepath} not found")
 
 
+def find_nested_requirements_txts(
+    base_requirements_txt: Path,
+) -> tuple[Path, set[Path]]:
+    """Recursively search the provided requirements.txt file to find nested
+    requirements files specified with `-r`.
+    """
+    sub_requirements_txt_files: set[Path] = set()
+    with base_requirements_txt.open("r") as f:
+        for line in f.readlines():
+            if line.startswith("-r"):
+                base, nested = find_nested_requirements_txts(
+                    base_requirements_txt.parent / line.split("-r")[1].strip(),
```

Comment:
In case there's `-r` inside a filename. Maybe you'll need another branch for `--requirement` too.

```suggestion
                    base_requirements_txt.parent / line.split("-r", 1)[1].strip(),
```